viadot/sources/spectrum.py

```diff
@@ -0,0 +1,166 @@
+            database (str): AWS Glue  catalog database name.
+            table (str): AWS Glue catalog table name.
+        """
+        table_location = wr.catalog.get_table_location(
+            boto3_session=self.session, database=database, table=table
+        )
+        wr.catalog.delete_table_if_exists(database=database, table=table)
+
+        wr.s3.delete_objects(path=table_location)
+
+    def from_df(
+        self,
+        df: pd.DataFrame,
+        to_path: str,
+        database: str,
+        table: str,
+        extention: str = ".parquet",
+        if_exists: Literal["overwrite", "append"] = "overwrite",
+        partition_cols: List[str] = None,
+        index: bool = False,
+        compression: str = None,
+        sep: str = ",",
+        description: str = "test",
+        columns_comments: Dict[str, str] = None,
+    ):
+        """
+        Upload a pandas `DataFrame` to a csv or parquet file.
+
+        Args:
+            df (pd.DataFrame): Pandas DataFrame
+            to_path (str): Path to a S3 folder. Defaults to None.
+            extention (str): Required file type.
+            database (str): AWS Glue  catalog database name.
+            table (str): AWS Glue catalog table name.
+            partition_cols (List[str]):
+            if_exists (str, optional): 'overwrite' to recreate any possible existing table or 'append' to keep any possible existing table. Defaults to overwrite.
+            index (bool, optional): Write row names (index). Defaults to False.
+            compression (str, optional): Compression style (None, snappy, gzip, zstd).
+            sep (str, optional): Field delimiter for the output file. Defaults to ','.
+            description (str, optional): Glue catalog table description.
+            columns_comments (Dict[str,str], optional) - Glue catalog columns names and the related comments.
+        """
+
+        if extention == ".parquet":
+            wr.s3.to_parquet(
+                boto3_session=self.session,
+                df=df,
+                path=to_path,
+                mode=if_exists,
+                index=index,
+                compression=compression,
+                dataset=True,
+                database=database,
+                table=table,
+                partition_cols=partition_cols,
+                description=description,
+                columns_comments=columns_comments,
+            )
+        elif extention == ".csv":
+            wr.s3.to_csv(
+                boto3_session=self.session,
+                df=df,
+                path=to_path,
+                dataset=True,
+                database=database,
+                table=table,
+                index=index,
+                sep=sep,
+            )
+        else:
+            raise ValueError("Only CSV and parquet formats are supported.")
+
+    def to_df(self, database: str = None, table: str = None):
+        """
+        Reads a parquet file to dataframe.
```

Comment:
I think the descr is incorrect; this reads a Spectrum table into a pandas DataFrame.

Reply:
Updated as per [🎨 improved structure and updated docstrings](https://github.com/dyvenia/viadot/pull/579/commits/13c71f9ed5e4afc46038950f56bea82ade0fd06c)